chiselFrontend/src/main/scala/chisel3/core/Reg.scala

```diff
@@ -153,12 +153,15 @@ object RegInit {
       requireIsChiselType(t, "reg type")
     }
     val reg = t.cloneTypeFull
-    val clock = Builder.forcedClock.ref
-    val reset = Builder.forcedReset.ref
+    val clock = Builder.forcedClock
+    val reset = Builder.forcedReset
+    if (reset.isInstanceOf[AsyncReset] && init.litOption.isEmpty) {
+      Builder.error(s"Register $this with async reset has non-literal reset value")
```

Comment:
If the Reset type is meant to be instantiable as both a Async or Sync reset, should this be a whitelist check (that the current reset of explicitly of a Bool type) instead?
Cons if it errors here: it's an over-eager error, you might never instantiate a non-literal reset value with an AsyncReset
Pros: you avoid a potential minefield of flakey code that uses non-literal reset values with Reset where the test cases don't try instantiating the module under AsyncReset

Reply:
Check is removed from Chisel into FIRRTL, checked after reset inference and supports async reset registers reset to aggregates.